resolwe/observers/protocol.py

```diff
@@ -22,3 +23,26 @@ class ChangeType(Enum):
 # or before and after a PermissionObject's container is changed.
 pre_permission_changed = dispatch.Signal(providing_args=["instance"])
 post_permission_changed = dispatch.Signal(providing_args=["instance"])
+
+# Signal to be sent before and after container is changed.
+pre_container_changed = dispatch.Signal(providing_args=["instance"])
+post_container_changed = dispatch.Signal(providing_args=["instance"])
+
+
+class ChannelsMessage(TypedDict):
+    """The type for channels message to be sent."""
+
+    type: str
+    object_id: Optional[int]
+    content_type_pk: int
+    change_type_value: int
+    source: Optional[tuple[str, int]]
+
+
+class WebsocketMessage(TypedDict):
+    """The type of websocket message to be sent."""
+
+    object_id: Optional[int]
+    change_type: str
+    source: Optional[tuple[str, int]]
+    subscription_id: str
```

Comment:
`object_id` is redundant, since it's already in `source`. Changing this will impact the functionality on the frontend.

Reply:
That's fair. Add a comment to explain this in code as well.

Reply:
Done, see protocol.